src/planner/parabolic/ParabolicUtil.cpp

```diff
@@ -0,0 +1,228 @@
+  transitionTimes.erase(startIt);
+
+  Eigen::VectorXd positionPrev, velocityPrev;
+  evaluateAtTime(*_inputPath, timePrev, positionPrev, velocityPrev);
+
+  auto _outputTrajectory = make_unique<aikido::trajectory::Spline>(_stateSpace,
+                                                          timePrev + _startTime);
+  auto segmentStartState = _stateSpace->createState();
+
+  for (const auto timeCurr : transitionTimes)
+  {
+    Eigen::VectorXd positionCurr, velocityCurr;
+    evaluateAtTime(*_inputPath, timeCurr, positionCurr, velocityCurr);
+
+    // Compute the spline coefficients for this segment of the trajectory. Each
+    // segment is expressed in the tangent space of the previous segment.
+    // TODO: We should apply the adjoint transform to velocity.
+    CubicSplineProblem problem(Vector2d(0, timeCurr - timePrev), 4, dimension);
+    problem.addConstantConstraint(0, 0, Eigen::VectorXd::Zero(dimension));
+    problem.addConstantConstraint(0, 1, velocityPrev);
+    problem.addConstantConstraint(1, 0, positionCurr - positionPrev);
+    problem.addConstantConstraint(1, 1, velocityCurr);
+    const auto spline = problem.fit();
+
+    // Forward-integrate the trajectory from the start in one step. This should
+    // prevent accumulation of numerical error over long trajectories. Note
+    // that this is only possible in real vector spaces and SO(2).
+    //_stateSpace->expMap(positionPrev, relativeState);
+    _stateSpace->expMap(positionPrev, segmentStartState);
+
+    // Add the ramp to the output trajectory.
+    assert(spline.getCoefficients().size() == 1);
+    const auto& coefficients = spline.getCoefficients().front();
+    _outputTrajectory->addSegment(
+      coefficients, timeCurr - timePrev, segmentStartState);
+
+    timePrev = timeCurr;
+    positionPrev = positionCurr;
+    velocityPrev = velocityCurr;
+  }
+
+  return _outputTrajectory;
+}
+
+std::unique_ptr<ParabolicRamp::DynamicPath>
+    convertToDynamicPath(const aikido::trajectory::Spline* _inputTrajectory,
+                         double& _startTime,
+                         const Eigen::VectorXd& _maxVelocity,
+                         const Eigen::VectorXd& _maxAcceleration)
+{
+  const auto stateSpace = _inputTrajectory->getStateSpace();
+  const auto numWaypoints = _inputTrajectory->getNumWaypoints();
+
+  std::vector<ParabolicRamp::Vector> milestones;
+  milestones.reserve(numWaypoints);
+
+  Eigen::VectorXd tangentVector;
+
+  for (size_t iwaypoint = 0; iwaypoint < numWaypoints; ++iwaypoint)
+  {
+     auto currentState = stateSpace->createState();
+     _inputTrajectory->getWaypoint(iwaypoint, currentState);
+
+     stateSpace->logMap(currentState, tangentVector);
+     milestones.emplace_back(toVector(tangentVector));
+  }
+
+  auto outputPath = make_unique<ParabolicRamp::DynamicPath>();
+  // Apply the adjoint limits
+  outputPath->Init(toVector(_maxVelocity), toVector(_maxAcceleration));
+  outputPath->SetMilestones(milestones);
```

Comment:
We need another version of this helper function - or an overload of this function - that optionally passes velocities to `SetMilestones`. We need that for all of the functions that accept non-piecewise linear splines.